src/Symfony/Component/Messenger/Transport/Serialization/Serializer.php

```diff
@@ -48,8 +49,17 @@ public function decode(array $encodedMessage)
      */
     public function encode($message): array
     {
+        $context = array();
+        if ($message instanceof WrappedMessage) {
+            /** @var SerializerContextAwareMessage|null $contextAwareMessage */
+            if ($contextAwareMessage = $message->unwrap(SerializerContextAwareMessage::class)) {
+                $context = $contextAwareMessage->getContext();
```

Comment:
I guess after https://github.com/symfony/symfony/pull/26941/files#diff-5e7347edce37c5886ec67b7ba02f3a9cR1019 both context entries can be merged? having priority the specific one over the general one?

Reply:
I would just replace actually. But that's debatable.